### ISSUE TITLE
ChangeDetectorRef `markForCheck` doesn't behave properly in Angular Elements

### ISSUE THREAD

**Issue by antch** [angular/angular#37981]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/elements

### Is this a regression?
No

### Description
Invoking `changeDetectorRef.markForCheck()` doesn't work properly inside of an Angular Element in some cases.
This issue surfaced when I tried to use [ng-select](https://github.com/ng-select/ng-select) within a Custom Element and noticed that external binding changes were not being propagated. I was able to the debug the issue and come up with a minimal reproduction.

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular-elements-markforcheck

In this example there is a native Angular and custom element implementation which both use a simple `ControlValueAccessor` child component w/ `OnPush` change detection. If `markForCheck()` is called in the component's `writeValue()` it updates the value in the native component but not the Custom Element.

Interesting to note, either of the following DO work:
```
    this.cd.detectChanges();
    this.zone.run(() => this.cd.markForCheck());
```

## 🔥 Exception or Error
None, values are just not updated properly.

## 🌍  Your Environment

**Angular Version:**
Not from `ng version`, but I pasted the dependencies from the StackBlitz:

<pre><code>
"dependencies": {
    "@angular/animations": "^10.0.1",
    "@angular/common": "^10.0.1",
    "@angular/compiler": "^10.0.1",
    "@angular/core": "^10.0.1",
    "@angular/elements": "^10.0.2",
    "@angular/forms": "^10.0.1",
    "@angular/platform-browser": "^10.0.1",
    "@angular/platform-browser-dynamic": "^10.0.1",
    "@angular/router": "^10.0.1",
    "@webcomponents/custom-elements": "^1.4.1",
    "@webcomponents/webcomponentsjs": "^2.4.3",
    "rxjs": "^6.5.5",
    "tslib": "^2.0.0",
    "zone.js": "^0.10.3"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.10.0",
    "@angular/cli": "9.0.0-rc.7",
    "@angular/compiler-cli": "9.0.0-rc.7",
    "@angular/language-service": "~7.0.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.11.0",
    "typescript": "~3.1.1"
  }
</code></pre>

**Anything else relevant?**
Also occurs in Angular 9.1.9.

**Comment by gkalpak** [comment]
I wonder if this will be fixed by #37814.

@antch, could you try out with the [build artifacts][1] from #37814? (See [docs/DEVELOPER.md][2] for details on how to use the build artifacts.)

[1]: https://circleci.com/gh/angular/angular/758316#artifacts/containers/0
[2]: https://github.com/angular/angular/blob/e50c7dcc4c11a61a63b8234cc9649cad7584ee76/docs/DEVELOPER.md#getting-packages-from-build-artifacts

**Comment by antch** [comment]
Unfortunately it does not look like that PR fixes the issue.

https://stackblitz.com/edit/angular-elements-markforcheck-with-37814?file=package.json

Same behavior as before -- if you click the "Clear" button under the Custom Element you don't see the null value reflected, until you click the button under the Native Element (or, theoretically, any other action which specifically triggers change detection).

**Comment by gkalpak** [comment]
@antch, I am afraig StackBlitz cannot install dependencies from `.tgz` archives (so it falls back to its default Angular version). So, I don't think it is using the artifacts at all. You need to export the project locally and try it.

Can you also try what happens if you include the Zone.js polyfill before the custom-element ones?

**Comment by antch** [comment]
Ah, you are correct. Running it locally it does appear that the issue is fixed with #37814. Thanks!

**Comment by gkalpak** [comment]
Smashing! Thank you for confirming, @antch ✨
